azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/NetworkSecurityRuleApiLiveTest.java

```diff
@@ -0,0 +1,214 @@
+   }
+
+   @Test(groups = "live", dependsOnMethods = "createNetworkSecurityRule")
+   public void getNetworkSecurityRule() {
+      final NetworkSecurityRule rule = createRule();
+      assertNotNull(rule);
+
+      final NetworkSecurityRuleApi ruleApi = api.getNetworkSecurityRuleApi(resourcegroup, DEFAULT_NSG_NAME);
+      NetworkSecurityRule result = ruleApi.get(rule.name());
+      assertNotNull(result);
+      assertNotNull(result.etag());
+      assertEquals(result.name(), rule.name());
+   }
+
+   @Test(groups = "live", dependsOnMethods = "createNetworkSecurityRule")
+   public void getNetworkSecurityDefaultRule() {
+      String defaultRuleName = "AllowVnetInBound";
+
+      final NetworkSecurityRuleApi ruleApi = api.getNetworkSecurityRuleApi(resourcegroup, DEFAULT_NSG_NAME);
+      NetworkSecurityRule result = ruleApi.getDefaultRule(defaultRuleName);
+
+      assertNotNull(result);
+      assertNotNull(result.etag());
+      assertEquals(result.name(), defaultRuleName);
+   }
+
+   @Test(groups = "live", dependsOnMethods = "createNetworkSecurityRule")
+   public void listNetworkSecurityRules() {
+      final NetworkSecurityRule rule = createRule();
+      assertNotNull(rule);
+
+      final NetworkSecurityRuleApi ruleApi = api.getNetworkSecurityRuleApi(resourcegroup, DEFAULT_NSG_NAME);
+      List<NetworkSecurityRule> result = ruleApi.list();
+
+      assertNotNull(result);
+      assertEquals(result.size(), 2);
+
+      boolean rulePresent = false;
+      for (int i = 0; i < result.size(); i++) {
+         if (result.get(i).name().equals(rule.name())) {
+            rulePresent = true;
+            break;
+         }
+      }
+      assertTrue(rulePresent);
+   }
+
+   @Test(groups = "live", dependsOnMethods = "createNetworkSecurityRule")
+   public void listDefaultSecurityRules() {
+      final NetworkSecurityRuleApi ruleApi = api.getNetworkSecurityRuleApi(resourcegroup, DEFAULT_NSG_NAME);
+      List<NetworkSecurityRule> result = ruleApi.listDefaultRules();
+
+      assertNotNull(result);
+      assertTrue(result.size() > 0);
+   }
+
+   @Test(groups = "live", dependsOnMethods = {"listNetworkSecurityRules", "listDefaultSecurityRules", "getNetworkSecurityRule"}, alwaysRun = true)
+   public void deleteNetworkSecurityRule() {
+      final NetworkSecurityRule rule = createRule();
+      assertNotNull(rule);
+
+      final NetworkSecurityRuleApi ruleApi = api.getNetworkSecurityRuleApi(resourcegroup, DEFAULT_NSG_NAME);
+      URI uri = ruleApi.delete(rule.name());
+      if (uri != null) {
+         assertTrue(uri.toString().contains("api-version"));
+         assertTrue(uri.toString().contains("operationresults"));
+
+         boolean jobDone = Predicates2.retry(new Predicate<URI>() {
+            @Override
+            public boolean apply(URI uri) {
+               return ParseJobStatus.JobStatus.DONE == api.getJobApi().jobStatus(uri);
+            }
+         }, 60 * 2 * 1000 /* 2 minute timeout */).apply(uri);
```

Comment:
This predicate is recurrent, and something that potentially could be reused. Please, bind the reusable predicates to the Guice context so the code can be properly shared in the different tests, but also in the provider code that needs to wait.

Take a look at how this kind of predicates are configured in the [DigitalOcean provider](https://github.com/jclouds/jclouds/blob/master/providers/digitalocean2/src/main/java/org/jclouds/digitalocean2/compute/config/DigitalOcean2ComputeServiceContextModule.java#L108-L216) and [how they are used in the live tests](https://github.com/jclouds/jclouds/blob/master/providers/digitalocean2/src/test/java/org/jclouds/digitalocean2/internal/BaseDigitalOcean2ApiLiveTest.java)